domain.go

```diff
@@ -168,6 +198,7 @@ type DomainDeviceList struct {
 	Inputs      []DomainInput      `xml:"input"`
 	Graphics    []DomainGraphic    `xml:"graphics"`
 	Videos      []DomainVideo      `xml:"video"`
+	Channels    []DomainChannel    `xml:"channel"`
```

Comment:
So internally libvirt treats  serial, console and channel devices as all using the same basic schema - chardevs - so what we should do here is fill out the DomainChardev struct with the missing pieces, rather than create DomainChannel